Documentation/writing-tizen-csharp-apps.md

```diff
@@ -0,0 +1,133 @@
+in_info.AddTensorInfo(TensorType.UInt8, new int[4] { 3, 224, 224, 1 });
+
+/* Output Dimension: 1001 for classification */
+TensorsData out_data = new TensorsInfo();
+out_info.AddTensorInfo(TensorType.UInt8, new int[4] { 1001, 1, 1, 1 });
+```
+
+<a name="load"></a>
+## Loading the neural network model and configures a runtime environment
+
+1. Since the model file is located at the resource directory for your own application, you need to get its absolute path as below.
+
+```C#
+string ResourcePath = Tizen.Applications.Application.Current.DirectoryInfo.Resource;
+string model_path = ResourcePath + "models/mobilenet_v1_1.0_224_quant.tflite";
+```
+
+2. Then you can load the neural network model from storage and configure runtime environment with `Tizen.MachineLearning.Inference.SingleShot` class. The first parameter is the absolute path to the neural network model file and remaining 2 are input/output TensorsInfo instance. `ArgumentException` maybe raise because of invalid parameter.
+
+```C#
+/* Create SingleShot instance with model information */
+SingleShot single = new SingleShot(model_path, in_info, out_info);
+```
+
+<a name="invoke"></a>
+## Invoking the neural network model with the given input data
+
+In order to invoke the model, you need to make the `Tizen.MachineLearning.Inference.TensorsData` instance to pass its input data. You can add the various types of Tensor data, which are already specified at `TensorInfo` instance. However, the maximum size of the TensorsData is 16. `IndexOutOfRangeException` could raise when exceeding the limit. Input data is passed as a byte array format (i.e. byte[]).
+
+```C#
+/* Input data for test */
+byte[] in_buffer = new byte[3 * 224 * 224 * 1];
+
+/* Set the input tensor data */
+TensorsData in_data = in_info.GetTensorsData();
+in_data.SetTensorData(0, in_buffer);
+```
+
+After preparing the input data, you can invoke the model and get the inference output result. `SingleShot.Invoke()` method gets input data to be inferred as parameter and  returns `Tizen.MachineLearning.InferenceTensorsData` instance which contains the inference result.
+
+```C#
+/* Invoke the model and get the inference result */
+TensorsData out_data = single.Invoke(in_data);
+```
+
+<a name="fatch"></a>
+## Fetching the inference result after invoking
+
+After calling `Invoke()` method of `Tizen.MachineLearning.Inference.SingleShot` class, 
+`Tizen.MachineLearning.Inference.TensorsData` instance is returned as the inference result.
+It can have multiple output data so you have to fetch each data by using `GetTensorData()` method. `IndexOutOfRangeException` could raise when exceeding the limit.
+
+```C#
+/* Get the first Tensor data from the inference result */
+byte[] out_buffer = out_data.GetTensorData(0);
+```
+
+As I have mentioned [above](#invoke), `TensorsData` class is used to send the Input data to a neural network model. Moreover, it provides the `Count` property for developers to get the number or Tensor it has.
```

Comment:
I am not sure about 'I' here. How about just 

> As mentioned [above](#invoke),

Reply:
ACK!